src/components/ui/Paginator/Paginator.tsx

```diff
@@ -0,0 +1,213 @@
+      const refSize = childrenSizes[i]
+
+      // if the current page would overflow after adding this element, push it to the array and start a new page
+      if (currentPageSize + refSize > parentSize && currentPageElements.length > 0) {
+        newPagesArray.push(currentPageElements)
+        currentPageSize = 0
+        currentPageElements = []
+      }
+
+      // if there is one dom element that is too long and would be scrollable, split it into separate pages
+      if (currentPageSize === 0 && refSize > parentSize) {
+        // split into pages and save where to scroll on changing page
+        let i = 0
+        while ((i + 1) * parentSize < refSize) {
+          let pageDom = ref.cloneNode(true) as HTMLDivElement
+
+          pageDom.setAttribute('scrollTo', String(i * parentSize))
+          newPagesArray.push([pageDom])
+          i++
+        }
+
+        // avoid having the last part of the element be very small, instead, make it the size of parent element
+        let lastPageDom = ref.cloneNode(true) as HTMLDivElement
+        lastPageDom.setAttribute('scrollTo', String(refSize - parentSize))
+        newPagesArray.push([lastPageDom])
+      } else if (currentPageSize + refSize <= parentSize) {
+        // if even after adding this element the page is not overflowing, add it to current page
+        currentPageElements.push(ref.cloneNode(true))
+        currentPageSize += refSize
+      }
+    })
+    // push leftover elements to last page if needed
+    if (currentPageElements.length > 0) {
+      newPagesArray.push(currentPageElements)
+    }
+
+    // If the screen was resized it might happen that there's now less pages when user was on the last page
+    if (currentPage > newPagesArray.length - 1) setCurrentPage(newPagesArray.length - 1)
+
+    setPages(newPagesArray)
+  }
+
+  useEffect(() => {
+    if (pages.length > 0 && pageRef.current !== null) {
+      // remove all elements of previous page
+      while (pageRef.current.firstChild) {
+        pageRef.current.removeChild(pageRef.current.firstChild)
+      }
+
+      // if there is only one element, and it has a scrollTo attribute, add that element and scroll it
+      if (
+        pages[currentPage].length == 1 &&
+        (pages[currentPage][0] as HTMLDivElement).getAttribute('scrollTo') !== null
+      ) {
+        const pageDiv = pages[currentPage][0] as HTMLDivElement
+        const scrollTo = pageDiv.getAttribute('scrollTo')
+
+        pageRef.current.appendChild(pageDiv)
+
+        pageRef.current.scrollTo({
+          left: orientation === 'horizontal' ? Number(scrollTo) : 0,
+          top: orientation === 'vertical' ? Number(scrollTo) : 0,
+          behavior: 'smooth',
+        })
+      } else {
+        // otherwise, just add all the current page elements
+        pages[currentPage].forEach((el: Node) => {
+          if (pageRef.current !== null) pageRef.current.appendChild(el)
+        })
+      }
+    }
+  }, [pages, currentPage, orientation])
```

Comment:
You should not add/remove child nodes using the vanilla JS methods, like `appendChild`, but rather use the children array passed directly, small example:
```javascript
<div>
  { childrenArray.slice(
    pages && pages[0] ? (currentPage > 0 ? pages[currentPage - 1].length : 0) : 0,
    pages && pages[0] ? (currentPage > 0 ? pages[currentPage - 1].length : 0) + (pages[currentPage].length) : childrenArray.length
  ).map((child, i) => (
    <div key={i}>
      {child}
    </div>
  ))}
</div>
```
This way only the child components in view are rendered, and their lifecycle methods also work.

Reply:
Ah yeah, I was initially doing this but running into many problems with scrolling, found out what was causing it and forgot to change that in the end.  No adding or removing doms now